dispatcher/MongoConnect.py

```diff
@@ -481,18 +489,19 @@ def process_commands(self):
         '''
         Process our internal command queue
         '''
+        sort = [('createdAt', 1)]
+        incoming = self.collections['command_queue']
+        outgoing = self.collections['outgoing_commands']
         while self.run == True:
             try:
-                next_cmd = self.collections['command_queue'].find_one({}, sort=[('createdAt', 1)])
-                if next_cmd is None:
+                if (next_cmd := incoming.find_one({}, sort=sort)) is None:
                     dt = 10
                 else:
-                    dt = (next_cmd['createdAt'] - datetime.datetime.utcnow()).total_seconds()
+                    dt = (next_cmd['createdAt'] - now()).total_seconds()
                 if dt < 0.01:
                     oid = next_cmd.pop('_id')
-                    ret = self.collections['outgoing_commands'].insert_one(next_cmd)
-                    self.collections['command_queue'].delete_one({'_id': oid})
-                    self.command_oid[next_cmd['detector']][next_cmd['command']] = ret.inserted_id
+                    ret = outgoing.insert_one(next_cmd)
```

Comment:
ret holds the ID not the document including next_cmd. That's what you want, I guess?